vllm/error_report.py

```diff
@@ -0,0 +1,159 @@
+from vllm.version import __version__ as VLLM_VERSION
+from vllm.worker.worker_base import ModelExecutionError
+
+logger = init_logger(__name__)
+
+
+# Hacky way to make sure we can serialize the object in JSON format
+def is_json_serializable(x):
+    try:
+        json.dumps(x)
+        return True
+    except (TypeError, OverflowError):
+        return False
+
+
+def prepare_object_to_dump(obj):
+    if isinstance(obj, dict):
+        return {k: prepare_object_to_dump(v) for k, v in obj.items()}
+    elif isinstance(obj, list):
+        return [prepare_object_to_dump(v) for v in obj]
+    elif isinstance(obj, set):
+        return [prepare_object_to_dump(v) for v in list(obj)]
+    elif isinstance(obj, tuple):
+        return [prepare_object_to_dump(v) for v in obj]
+    elif isinstance(obj, enum.Enum):
+        return repr(obj)
+    elif isinstance(obj, SequenceData):
+        # Custom representation (based on SequenceData.__repr__)
+        # to obfuscate some parameters
+        return {
+            "class": "SequenceData",
+            "prompt_token_ids_len": len(obj._prompt_token_ids),
+            "output_token_ids_len": len(obj.output_token_ids),
+            "cumulative_logprob": obj.cumulative_logprob,
+            "get_num_computed_tokens": obj.get_num_computed_tokens()
+        }
+
+    elif isinstance(obj, NewRequestData):
+        obj_dict: Dict[str, Any] = {'class': type(obj).__name__}
+        for k, v in obj.__dict__.items():
+            if k == 'prompt_token_ids':
+                obj_dict['prompt_token_ids_len'] = len(v)
+            elif k == 'prompt':
+                obj_dict['prompt'] = ""
+            else:
+                obj_dict[k] = prepare_object_to_dump(v)
+
+        return obj_dict
+    elif isinstance(obj, torch.Tensor):
+        # We only print the 'draft'of the tensor to not expose sensitive data
+        # and to get some metadata in case of CUDA illegal memory access
+        return (f"Tensor(shape={obj.shape}, "
+                f"device={obj.device},"
+                f"dtype={obj.dtype})")
+    elif hasattr(obj, '__dict__'):
+        obj_dict = {'class': type(obj).__name__}
+        obj_dict.update(obj.__dict__)
+        return prepare_object_to_dump(obj_dict)
+    else:
+        # Try to make sure we can serialize the object
+        # to avoid exception
+        if is_json_serializable(obj):
+            return obj
+        else:
+            return repr(obj)
+
+
+def dump_engine_exception(err: BaseException,
+                          config: VllmConfig,
+                          engine_version: int,
+                          stats: Union[Stats, None] = None,
+                          use_cached_outputs: Union[bool, None] = None,
+                          execute_model_req: Union[ExecuteModelRequest,
+                                                   None] = None):
+
+    assert engine_version == 0 or engine_version == 1
+
+    logger.error("Engine crashed, dumping input data")
```

Comment:
nit: "Engine crashed" is not the business of this function. Technically you can still call this function even engine is not crashed, so probably just `Dumping input data`